src/Generation/TestNameValueProducer.php

```diff
@@ -145,7 +147,8 @@ public function fieldInit(MethodDetails $method, FieldDetails $field, &$fieldVar
 
         // This should only use oneof wrapper types if the oneof is on the top level request message.
         $inTopLevel = $method->inputMsg === $field->containingMessage;
-        if ($field->isOneOf && $inTopLevel) {
+        // This should only be used in V1 tests, as V2 does not need Oneof wrappers.
+        if ($field->isOneOf && $inTopLevel && !$this->v2) {
```

Comment:
I don't think we need this `$v2` variable, we can check the `$method->serviceDetails->migrationMode`

Reply:
The logic isn't dependent on Migration Mode, it depends on whether or not you're _using_ the V1 or V2 client. Since some migration modes have both, the only way to know is to explicitly pass it in.

I changed the variable name to `$useOneofWrappers` instead but I'm not sure that's better. I am definitely open to any better ways to do this.

Reply:
I could potentially change it to `$isV2Test`... I think that might make it the most clear

Reply:
Oh ok thanks!